### DISCUSSION TITLE
The values of Q_i, e_i , D_i and \kappa_i for the last spatial node i=n+1.

### DISCUSSION THREAD

**Discussion by Srishti-Siddharth-iitb** [GazzolaLab/PyElastica#419] [Q&A]
I am trying to simulate a single Cosserat rod in MATLAB using your codes and the numerical scheme discussed in the following 2018 paper: https://doi.org/10.1098/rsos.171628. To stimulate the discrete equations (3.8) and (3.9) of the paper,  the values of Q_i, e_i, D_i and \kappa_i are needed for the last spatial node i.e. i=n+1. However, according to equations (3.3), (3.4) and (3.5) of the paper, the above quantities are not described for i=n+1. I wanted to understand how these quantities have been computed in your Python code. It would be helpful if you could provide the line number in the Python code which can be used as a reference to look at the computation of these quantities for the last spatial node. Also, do ghost nodes play a role here?

**Comment by armantekinalp** [comment]
You can take a look to this [file](https://github.com/GazzolaLab/PyElastica/blob/master/elastica/rod/cosserat_rod.py) to see how rod equations are computed.  I am also moving this issue to discussion.

**Reply by Srishti-Siddharth-iitb** [reply]
Thank you, I will take a look at it.
